src/transcription/prerecorded/audio_source.rs

```diff
@@ -2,67 +2,69 @@ use reqwest::{header::CONTENT_TYPE, RequestBuilder};
+pub struct AudioSource(InternalAudioSource);
 
-    pub trait Sealed {}
-
-    impl Sealed for &UrlSource<'_> {}
-    impl<B: Into<reqwest::Body>> Sealed for BufferSource<'_, B> {}
+#[derive(Debug)]
+enum InternalAudioSource {
+    Url(String),
+    Buffer {
+        buffer: reqwest::Body,
+        mime_type: Option<String>,
+    },
 }
 
-/// Used as a parameter for [`Transcription::prerecorded`](crate::transcription::Transcription::prerecorded) and similar functions.
-///
-/// Instructs Deepgram to download the audio from the specified URL.
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Serialize)]
-pub struct UrlSource<'a> {
-    /// URL of audio file to transcribe.
-    pub url: &'a str,
-}
+impl AudioSource {
+    /// Constructs an [`AudioSource`] that will instruct Deepgram to download the audio from the specified URL.
+    pub fn from_url(url: &str) -> Self {
+        Self(InternalAudioSource::Url(String::from(url)))
+    }
 
-/// Used as a parameter for [`Transcription::prerecorded`](crate::transcription::Transcription::prerecorded) and similar functions.
-///
-/// Uploads the raw binary audio data to Deepgram as part of the request.
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
-pub struct BufferSource<'a, B: Into<reqwest::Body>> {
-    /// The source of the raw binary audio data, such as a [`tokio::fs::File`].
+    /// Constructs an [`AudioSource`] that will upload the raw binary audio data to Deepgram as part of the request.
     ///
-    /// It can be any type that implements [`Into<reqwest::Body>`].
+    /// The buffer can be any type that implements [`Into<reqwest::Body>`], such as a [`tokio::fs::File`].
     /// See [trait implementations for `reqwest::Body`](reqwest::Body#trait-implementations)
     /// for a list of types that already implement it.
-    pub buffer: B,
-
-    /// Optionally specify the [MIME type][mime] of the raw binary audio data.
+    ///
+    /// Use [`AudioSource::from_buffer_with_mime_type`] if you want to specify a [MIME type][mime].
     ///
     /// [mime]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#audio_and_video_types
-    pub mimetype: Option<&'a str>,
-}
+    pub fn from_buffer(buffer: impl Into<reqwest::Body>) -> Self {
+        Self(InternalAudioSource::Buffer {
+            buffer: buffer.into(),
+            mime_type: None,
+        })
+    }
 
-impl AudioSource for &UrlSource<'_> {
-    fn fill_body(self, request_builder: RequestBuilder) -> RequestBuilder {
-        request_builder.json(self)
+    /// Same as [`AudioSource::from_buffer`], but allows you to specify a [MIME type][mime].
+    ///
+    /// [mime]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#audio_and_video_types
+    pub fn from_buffer_with_mime_type(buffer: impl Into<reqwest::Body>, mime_type: &str) -> Self {
+        Self(InternalAudioSource::Buffer {
+            buffer: buffer.into(),
+            mime_type: Some(String::from(mime_type)),
```

Comment:
Open for discussion: Is it worth using the [`mime::Mime`](https://docs.rs/mime/0.3.16/mime/struct.Mime.html) type for this?

Reply:
docs could provide an example like
```rust
let source = AudioSource::from_buffer_with_mimetype(buffer, "audio/mp3".parse()?);
````

Reply:
I think this would make sense if the majority of users coming to this part already had a `Mime` object or if we were doing something with it other than just turning it back into a string.

It doesn't eliminate the possibility of them getting the MIME type wrong (for example, `"auido/wav"` parses without error). It would just catch some typos before the network request rather than after.

I don't feel like this is worth making using the SDK more complicated.